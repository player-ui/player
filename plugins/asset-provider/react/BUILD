load("//:index.bzl", "javascript_pipeline")

javascript_pipeline(
    name = "@player-ui/asset-provider-plugin-react",
    dependencies = [
        "//react/subscribe:@player-ui/react-subscribe",
    ],
    peer_dependencies = [
        "//react/player:@player-ui/react",
    ],
)
