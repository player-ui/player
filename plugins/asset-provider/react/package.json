{
  "name": "@player-ui/asset-provider-plugin-react",
  "version": "0.0.0-PLACEHOLDER",
  "main": "src/index.tsx",
  "dependencies": {
    "@player-ui/react-subscribe": "workspace:*"
  },
  "peerDependencies": {
    "@player-ui/react": "workspace:*"
  }
}
