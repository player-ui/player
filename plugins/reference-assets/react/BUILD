load("//:index.bzl", "javascript_pipeline")

javascript_pipeline(
    name = "@player-ui/reference-assets-plugin-react",
    dependencies = [
        "//plugins/reference-assets/core:@player-ui/reference-assets-plugin",
        "//core/partial-match-registry:@player-ui/partial-match-registry",
        "//plugins/asset-provider/react:@player-ui/asset-provider-plugin-react",
        "//plugins/beacon/react:@player-ui/beacon-plugin-react",
        "@npm//clsx",
        "@npm//@chakra-ui/react",
        "@npm//@chakra-ui/icons"
    ],
    peer_dependencies = [
        "@npm//@types/node",
        "//react/player:@player-ui/react",
    ],
    test_data = [
        "@npm//@testing-library/react",
        "//core/make-flow:@player-ui/make-flow",
    ],
    xlr_mode = "plugin"
)

filegroup(
    name = "stories",
    srcs = glob(["src/**/*.stories.tsx"])
)