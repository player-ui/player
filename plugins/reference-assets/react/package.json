{
  "name": "@player-ui/reference-assets-plugin-react",
  "version": "0.0.0-PLACEHOLDER",
  "main": "src/index.tsx",
  "dependencies": {
    "@player-ui/asset-provider-plugin-react": "workspace:*",
    "@player-ui/beacon-plugin-react": "workspace:*",
    "@player-ui/partial-match-registry": "workspace:*",
    "@player-ui/reference-assets-plugin": "workspace:*"
  },
  "devDependencies": {
    "@player-ui/make-flow": "workspace:*"
  },
  "peerDependencies": {
    "@player-ui/react": "workspace:*"
  }
}
