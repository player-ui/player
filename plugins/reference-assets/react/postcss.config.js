/* eslint-disable @typescript-eslint/no-var-requires */
module.exports = {
  plugins: [require("tailwindcss")(), require("autoprefixer")()],
};
