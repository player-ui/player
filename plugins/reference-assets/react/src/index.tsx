import "./global.css";

export * from "./plugin";
export * from "./assets";
