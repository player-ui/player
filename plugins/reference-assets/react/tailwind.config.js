import { fontFamily } from "tailwindcss/defaultTheme";

/** @type {import('tailwindcss').Config} */
export default {
  prefix: "player-",
  darkMode: ["class"],

  content: ["src/**/*.{ts,tsx}"],
  theme: {
    container: {
      center: true,
      padding: "2rem",
      screens: {
        "2xl": "1400px",
      },
    },
    extend: {
      colors: {
        border: "hsl(var(--player-border))",
        input: "hsl(var(--player-input))",
        ring: "hsl(var(--player-ring))",
        background: "hsl(var(--player-background))",
        foreground: "hsl(var(--player-foreground))",
        primary: {
          DEFAULT: "hsl(var(--player-primary))",
          foreground: "hsl(var(--player-primary-foreground))",
        },
        secondary: {
          DEFAULT: "hsl(var(--player-secondary))",
          foreground: "hsl(var(--player-secondary-foreground))",
        },
        destructive: {
          DEFAULT: "hsl(var(--player-destructive))",
          foreground: "hsl(var(--player-destructive-foreground))",
        },
        muted: {
          DEFAULT: "hsl(var(--player-muted))",
          foreground: "hsl(var(--player-muted-foreground))",
        },
        accent: {
          DEFAULT: "hsl(var(--player-accent))",
          foreground: "hsl(var(--player-accent-foreground))",
        },
        popover: {
          DEFAULT: "hsl(var(--player-popover))",
          foreground: "hsl(var(--player-popover-foreground))",
        },
        card: {
          DEFAULT: "hsl(var(--player-card))",
          foreground: "hsl(var(--player-card-foreground))",
        },
      },
      borderRadius: {
        lg: `var(--player-radius)`,
        md: `calc(var(--player-radius) - 2px)`,
        sm: "calc(var(--player-radius) - 4px)",
      },
      fontFamily: {
        sans: ["var(--player-font-sans)", ...fontFamily.sans],
      },
      keyframes: {
        "player-accordion-down": {
          from: { height: "0" },
          to: { height: "var(--player-radix-accordion-content-height)" },
        },
        "player-accordion-up": {
          from: { height: "var(--player-radix-accordion-content-height)" },
          to: { height: "0" },
        },
      },
      animation: {
        "accordion-down": "player-accordion-down 0.2s ease-out",
        "accordion-up": "player-accordion-up 0.2s ease-out",
      },
    },
    corePlugins: {
      preflight: false,
    },
  },
  plugins: [require("tailwindcss-animate")],
};
