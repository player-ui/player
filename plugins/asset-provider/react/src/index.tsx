import React from 'react';
import type { ReactPlayer, ReactPlayerPlugin } from '@player-ui/react';
import { AssetContext } from '@player-ui/react';

export type AssetRegistryEntries = Array<[any, React.ComponentType<any>]>;

/**
 * A streamlined way of registering custom assets with the web-player
 */
export class AssetProviderPlugin implements ReactPlayerPlugin {
  name = 'web-asset-provider-plugin';

  private readonly entries: AssetRegistryEntries;

  /**
   * Create a plugin that will register the given React components as assets to Player
   *
   * @param entries - A list of [match, Component] tuples.
   * If the [match] is an object, it is used as-is.  If it's a `string`, we assume the match represents the 'type' property ({ type: match })
   */
  constructor(entries: AssetRegistryEntries) {
    this.entries = entries;
  }

  applyReact(rp: ReactPlayer) {
    this.entries.forEach(([match, comp]) => {
      const normalizedMatch =
        typeof match === 'string' ? { type: match } : match;

      rp.assetRegistry.set(normalizedMatch, comp);
    });

    // Because some instances may end up with a different copy of the `AssetContext` (depending on bundling and such)
    // We add an entry to use the local version of `@web-player/asset` -- but still utilize the same rp.assetRegistry
    rp.hooks.webComponent.tap(this.name, (Comp) => {
      return () => (
        <AssetContext.Provider value={{ registry: rp.assetRegistry }}>
          <Comp />
        </AssetContext.Provider>
      );
    });
  }
}
