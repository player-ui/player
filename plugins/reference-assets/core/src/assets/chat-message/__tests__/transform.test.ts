import { describe, it, expect, vitest } from "vitest";
import { runTransform } from "@player-ui/asset-testing-library";
import { chatMessageTransform } from "..";
import {
  AsyncNodePlugin,
  AsyncNodePluginPlugin,
} from "@player-ui/async-node-plugin";
import { waitFor } from "@testing-library/react";

describe("chatMessage transform", () => {
  it("generates a new multi-node with async node placeholder", async () => {
    const plugin = new AsyncNodePlugin({
      plugins: [new AsyncNodePluginPlugin()],
    });

    const test = vitest.fn();

    plugin.hooks.onAsyncNode.tap("test", async (node) => test());

    const ref = runTransform(
      "chat-message",
      chatMessageTransform,
      {
        type: "chat-message",
        id: "1",
        value: {
          asset: {
            id: "1",
            type: "text",
            value: "chat message",
          },
        },
      },
      [plugin],
    );

    /**
     * Check if async node exists in ASTMap after transform and resolve
     * onAsyncNode hook is only called when there is async node
     */
    await waitFor(() => {
      expect(test).toBeCalled();
    });

    expect(ref.current).toMatchObject({
      id: expect.any(String),
      type: "collection",
      values: [
        {
          asset: {
            id: "1",
            type: "text",
            value: "chat message",
          },
        },
      ],
    });

    expect(test).toHaveBeenCalled();
  });
});
