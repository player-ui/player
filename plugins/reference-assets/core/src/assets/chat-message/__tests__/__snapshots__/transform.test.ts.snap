// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`chatMessage transform > generates a new multi-node with async node placeholder 1`] = `
{
  "id": "chat-message-wrapper",
  "type": "chat-message-wrapper",
  "values": [
    {
      "id": "1",
      "type": "text",
      "value": "Hello World!",
    },
  ],
}
`;
